File: library/library/report/books_details/books_details.py
# Copyright (c) 2013, Bala Murugan and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
from frappe import _
import frappe

def execute(filters=None):
	#columns, data = [], []
	return get_columns(), get_data(filters)

def get_data(filters):
    print(f"\n\n\n\{filters}\n\n\n\n")
    #data = frappe.db.sql("""SELECT name,book_name,author,publisher,book_rent,status FROM `tabBooks`;""")
    data = frappe.db.sql("""SELECT name,book_name,author,publisher,book_rent,status, COUNT(book_rent) OVER (PARTITION BY book_name) AS total_book  FROM `tabBooks`;""")
    return data
   
def get_columns():
    return[
		"ISBN:Link/Books:150",
		"Book Name:Data:250",
		"Book Author:Data:150",
		"Book Publisher:Data:150",
		"Book Rent:Data:150",
		"Book Status:Selection:150",
		"Total Book:150"
	]

File: library/library/report/books_details/books_details.js
// Copyright (c) 2016, Bala Murugan and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Books Details"] = {
	"filters": [
		{
			"fieldname": "book_name",
			"label":__("Book Name"),
			"fieldtype": "Data",
			"width": 100,
			"reqd": 0,
		},
		{
			"fieldname": "status",
			"label":__("Book Status"),
			"fieldtype": "Select",
			"options":['Issued','Available'],
			"default":"",
			"width": 100,
			"reqd": 0,
		}

	]
};
